src/ToricVarieties/NormalToricVarieties/attributes.jl

```diff
@@ -313,6 +313,131 @@ end
 export map_from_weil_divisors_to_class_group
 
 
+@doc Markdown.doc"""
+    map_from_cartier_divisor_group_to_torus_invariant_divisor_group(v::AbstractNormalToricVariety)
+
+Computes the embedding of the group of Cartier divisors into the group of
+torus-invariant Weil divisors of an abstract normal toric variety `v`.
+
+# Examples
+```jdoctest
+julia> p2 = toric_projective_space(2)
+A normal toric variety corresponding to a polyhedral fan in ambient dimension 2
+
+julia> map_from_cartier_divisor_group_to_torus_invariant_divisor_group(p2)
+Map with following data
+Domain:
+=======
+Abelian group with structure: Z^3
+Codomain:
+=========
+Abelian group with structure: Z^3
+```
+"""
+function map_from_cartier_divisor_group_to_torus_invariant_divisor_group(v::AbstractNormalToricVariety)
+    # check input
+    if hastorusfactor(v)
+        throw(ArgumentError("Group of the torus-invariant Cartier divisors can only be computed if the variety has no torus factor."))
+    end
+    
+    # identify rays and cones
+    rays = Polymake.common.primitive(pm_object(v).RAYS)
+    max_cones = incidence_matrix(maximal_cones(fan(v)))
+    number_of_rays = size(rays)[1]
+    number_of_cones = size(max_cones)[1]
+    
+    # compute quantities needed to construct the matrices
+    rc = rank(character_lattice(v))
+    number_ray_is_part_of_max_cones = [length(max_cones[:,k].s) for k in 1:number_of_rays]
+    s = sum(number_ray_is_part_of_max_cones)
+    cones_ray_is_part_of = [filter(x -> max_cones[x,r], 1:number_of_cones) for r in 1:number_of_rays]
+    
+    # compute the matrix for the scalar products
+    map_for_scalar_products = zero_matrix(ZZ, number_of_cones * rc, s)
+    column = 1
+    for i in 1:number_of_rays
+        for j in cones_ray_is_part_of[i]
+            map_for_scalar_products[(j-1)*rc+1:j*rc, column] = [fmpz(c) for c in rays[i,:]]
+            column += 1
+        end
+    end
+    
+    # compute the matrix for differences
+    map_for_difference_of_elements = zero_matrix(ZZ, s, s-number_of_rays)
+    row = 1
+    col = 1
+    for i in 1:number_of_rays
+        ncol = number_ray_is_part_of_max_cones[i]-1
+        map_for_difference_of_elements[row, col:(col+ncol-1)] = [fmpz(1) for c in 1:ncol]
```

Comment:
```suggestion
        map_for_difference_of_elements[row, col:(col+ncol-1)] = fill(fmpz(1), ncol)
```
This is more efficient but it also means that all entries of the vector are *identical* (not just equal), which can cause problems if you do in-place modifications of those `fmpz` values... which you are not doing